content/blog/2020-11-16-reproducibility-in-ml-my-tools/index.en.md

```diff
@@ -0,0 +1,122 @@
+
+In this post, 
+
+- I will define "reproducible machine learning" and explain why it matters,
+- I will give three problems related to reproducibility that I've personally faced and why solving them is essential, 
+- I will give solutions I've tried developing on my own to solve these problems and reasons why these solutions are not suited for the problem, and
+- I will give solutions I'm now using to solve these problems and the reason behind these choices.
+
+## Reproducible Machine Learning
+In machine learning, reproducibility corresponds either to reproduce results, or to obtain similar results by re-executing a source code ([Pineau et al. 2020](https://arxiv.org/abs/2003.12206)).
+        
+It means our solution needs to be shareable among peers, and the results that we claim need to be reproducible. From a practical point of view, this translates into (1) being able to deploy our model into production and (2) that we have confidence that our predictions are "accurate" (i.e. performance will not drastically decrease when in production).
+
+## Managing results
+I've faced the problem of managing my results properly during my first project. Like any naive person, instead of reading the user manual "How to manage any results like a champ," I brute-forced my way to it by creating `.txt` files of my results. 
+
+It seemed nice: all I add to do was create a "meaningful" filename, such as `param_1_param_2_..._param_100.txt` plus a timestamp and dump all the results. I was so naive: at one point, I've got more than 100 files for that project only. Now, try finding which experiment was the best. 
+
+The problem with this approach is that it is complicated to manage all those files. Indeed, it is nearly impossible to be efficient when comparing all the results since that every time you create a new file, you need to go through all your results again. Also, sharing this kind of work between team members is pure insanity.
+
+Possible solutions for that specific problem should (1) efficiently allow a user to compare results between different runs or parameters, (2) easily save the results and (3) not add too much overhead to use. Before giving any good solution, let's discuss the second problem since the proposed solution also solves this problem.
```

Comment:
Peut-être ça change trop le meaning.
```suggestion
Possible solutions for that specific problem should (1) efficiently allow a user to compare results between different runs or parameters, (2) easily save the results and (3) not add too much overhead to the process. Before giving any good solution, let's discuss the second problem since the proposed solution also solves this problem.
```